sdk/data/azcosmos/cosmos_container.go

```diff
@@ -399,6 +401,59 @@ func (c *ContainerClient) DeleteItem(
 	return newItemResponse(azResponse)
 }
 
+// QueryItems executes a single partition query in a Cosmos container.
+// ctx - The context for the request.
+// query - The SQL query to execute.
+// partitionKey - The partition key to scope the query on.
+// o - Options for the operation.
+func (c *ContainerClient) QueryItems(query string, partitionKey PartitionKey, o *QueryOptions) *runtime.Pager[QueryItemsResponse] {
+	correlatedActivityId, _ := uuid.New()
+	h := headerOptionsOverride{
+		partitionKey:         &partitionKey,
+		correlatedActivityId: &correlatedActivityId,
+	}
+
+	if o == nil {
+		o = &QueryOptions{}
+	}
+
+	operationContext := pipelineRequestOptions{
+		resourceType:          resourceTypeDocument,
+		resourceAddress:       c.link,
+		headerOptionsOverride: &h,
+	}
+
+	path, _ := generatePathForNameBased(resourceTypeDocument, operationContext.resourceAddress, true)
+
+	return runtime.NewPager(runtime.PageProcessor[QueryItemsResponse]{
+		More: func(page QueryItemsResponse) bool {
+			return page.ContinuationToken != ""
```

Comment:
Not needed for input string parameters I believe. You can simply test the property is an empty string or not without having to add the extra complexity of a pointer. For properties that are of a different type (like a the ConsistencyLevel) where you need to test if the value was specified (because the default is actually a valid value and not a "not set"), pointers help.